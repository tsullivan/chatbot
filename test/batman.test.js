const request = require('supertest');
const { getServer } = require('../server');
const { utilFactory } = require('./utils');

const app = getServer();
const agent = request.agent(app);
const { handshake } = utilFactory(agent);

const testFields = body => ({
  message: body.message,
  format: body.format,
});

const sendTest = message =>
  agent
    .post('/chat')
    .send({ format: 'user', message })
    .expect(200);

describe('batman', () => {
  test('should win at the batman game', async () => {
    await handshake();

    const { body: a1 } = await sendTest('play batman');
    expect(testFields(a1)).toMatchSnapshot();

    const { body: b1 } = await sendTest('batmobile');
    expect(testFields(b1)).toMatchSnapshot();

    const { body: c1 } = await sendTest('77');
    expect(testFields(c1)).toMatchSnapshot();
  });

  test('should lose at the batman game', async () => {
    await handshake(app);

    const { body: a2 } = await sendTest('play batman');
    expect(testFields(a2)).toMatchSnapshot();

    const { body: b2 } = await sendTest('batgirl');
    expect(testFields(b2)).toMatchSnapshot();

    const { body: c2 } = await sendTest('kl');
    expect(testFields(c2)).toMatchSnapshot();
  });
});
