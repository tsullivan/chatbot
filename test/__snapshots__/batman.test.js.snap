// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`batman should lose at the batman game 1`] = `
Object {
  "format": "markdown",
  "message": "Let's play Batman. Type in stuff for the Batcave. Type \\"77\\" to keep your points or \\"kl\\" to throw it all away.",
}
`;

exports[`batman should lose at the batman game 2`] = `
Object {
  "format": "markdown",
  "message": "Adding batgirl to the Batcave.",
}
`;

exports[`batman should lose at the batman game 3`] = `
Object {
  "format": "markdown",
  "message": "Type \\"kl\\" and you lose the Batgame :( \\"kl\\" ended the Batgame. Your Batstuff is:

- batgirl

Your Batscore is 0.
That was fun, Tim!",
}
`;

exports[`batman should win at the batman game 1`] = `
Object {
  "format": "markdown",
  "message": "Let's play Batman. Type in stuff for the Batcave. Type \\"77\\" to keep your points or \\"kl\\" to throw it all away.",
}
`;

exports[`batman should win at the batman game 2`] = `
Object {
  "format": "markdown",
  "message": "Adding batmobile to the Batcave.",
}
`;

exports[`batman should win at the batman game 3`] = `
Object {
  "format": "markdown",
  "message": "You won! Right now!! \\"77\\" ended the Batgame. Your Batstuff is:

- batmobile

Your Batscore is 1.
That was fun, Tim!",
}
`;
